tracim/views/core_api/schemas.py

```diff
@@ -355,3 +382,129 @@ class ContentDigestSchema(marshmallow.Schema):
                     'for sub-contents. Default is True. '
                     'In first version of the API, this field is always True',
     )
+
+
+#####
+# Content
+#####
+
+class ContentSchema(ContentDigestSchema):
+    current_revision_id = marshmallow.fields.Int(example=12)
+    created = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='Content creation date',
+    )
+    author = marshmallow.fields.Nested(UserDigestSchema)
+    modified = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='date of last modification of content',
+    )
+    last_modifier = marshmallow.fields.Nested(UserDigestSchema)
+
+
+class ThreadContentSchema(ContentSchema):
+    content_type = marshmallow.fields.Str(
+        example='thread',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('Description of Thread')
+
+
+class HtmlDocumentContentSchema(ContentSchema):
+    content_type = marshmallow.fields.Str(
+        example='html-content',
```

Comment:
invalid example

Reply:
if you distinguish `ThreadContentSchema` and `HtmlDocumentContentSchema` then you must hard-write the content_type as it can't be for example `file`.

The right way to do this is probably to define a unique schema : 

```
TextBasedContentSchema(ContentSchema):
    content_type = marshmallow.fields.Str(
        example='html-content',
        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
    )
    raw_content = marshmallow.fields.String('Content of the object, may be raw text or <b>html</b> for example')
```

The reason for that : when will deal with markdwon, SQL, or any other content, we won't need a new schema definition

Reply:
hard_write of content_type is already done here : https://github.com/tracim/tracim_backend/pull/93/commits/60d9bb08b354557bf63ee4f594d0b1be1b54c70f .
If i want to modify heavily schema organisation, i prefer to do it in this branch .